scripts/scil_find_script.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Search through all of SCILPY scripts and their docstrings. The output of the
+search will be the intersection of all provided keywords, found either in the
+script name or in its docstring.
+"""
+
+from argparse import ArgumentParser, RawTextHelpFormatter
+import ast
+import inspect
+import pathlib
+import re
+
+import numpy as np
+import scilpy
+from scilpy.io.utils import add_verbose_arg
+
+RED = '\033[31m'
+BOLD = '\033[1m'
+ENDC = '\033[0m'
+
+
+def _build_arg_parser():
+    parser = ArgumentParser(description=__doc__,
+                            formatter_class=RawTextHelpFormatter)
+    parser.add_argument('keywords', nargs='+',
+                        help="Search keywords")
+    return parser
+
+
+def main():
+    parser = _build_arg_parser()
+    add_verbose_arg(parser)
+    args = parser.parse_args()
+
+    scilpy_init_file = inspect.getfile(scilpy)  # scilpy/scilpy/__init__.py
+    script_dir = pathlib.Path(scilpy_init_file).parent / "../scripts"
+    matches = []
+
+    kw_subs = [re.compile("(" + re.escape(kw) + ")", re.IGNORECASE)
+               for kw in args.keywords]
+
+    for script in sorted(script_dir.glob('*.py')):
+        filename = script.name
+        docstring = _get_docstring(str(script))
+
+        if args.verbose:
+            # Display full docstring
+            display_info = docstring
+        else:
+            # Extract first sentence by finding the first dot
+            display_info = _extract_first_sentence(docstring)
+            display_info = display_info.replace("\n", " ")
+
+        # Test intersection of all keywords, either in filename or docstring
+        if not _is_matching_keywords(args.keywords, [filename, docstring]):
+            continue
+
+        matches.append(filename)
+
+        # new_key calls regex group \1 to keep the same case
+        new_key = "{}\\1{}".format(RED + BOLD, ENDC)
+        for regex in kw_subs:
+            filename = regex.sub(new_key, filename)
+            display_info = regex.sub(new_key, display_info)
+
+        display_info = display_info or "No docstring available!"
+
+        if args.verbose:
+            print("=====", filename, "====")
+            print('"{}"'.format(display_info))
+            print()
+        else:
+            print(filename, '"{}"'.format(display_info))
+
+    if not matches:
+        print("No results found!")
+
+
+def _is_matching_keywords(keywords, texts):
+    """Test multiple texts for matching keywords. Returns True only if all
+    keywords are present in any of the texts.
+
```

Comment:
I'm not sure "is" is the good name.

_find?

Reply:
It doesn't return aything, son I don't think '_find' is the right one. What do you think of '_test_matching_keywords'?